### DISCUSSION TITLE
add support for multiple media values in one property

### DISCUSSION THREAD

**Discussion by seeden** [tamagui/tamagui#71] [General]
I would like to use properties with simplified multiple media values

```js
<Component
width={{ base: '50%', md: '43%', lg: '35%', xl: '28%' }}
/>
```

instead of

```js
<Component
  width="50%"
  $gtSm={{ width: '50%' }}
  $gtMd={{ width: '43%' }}
  $gtLg={{ width: '35%' }}
  $gtXl={{ width: '28%' }}
/>
```

Do you have any suggestions on how to achieve it?
Thank you

**Comment by natew** [comment]
I've spent a lot of time thinking about this and I think it's an either/or and I like the trade-off.

The inline (I'll call it that) media query syntax you've suggested is popular and I get the appeal, but ultimately I think it leads to confusing results once you go beyond even one or two properties, especially once you combine with psuedo queries and format with prettier, which will break lines up inconsistently.

Here's some made up styles but you get the point:

```tsx
export default () => {
  return (
    <Nested>
      <Component
        width={{ base: '50%', md: '43%', lg: '35%', xl: '28%' }}
        flexDirection={{ base: 'row', md: 'column', lg: 'column-reverse', xl: 'row-reverse' }}
        backgroundColor={{
          base: '$background',
          md: '$backgroundStronger',
          lg: '$backgroundStrongest',
        }}
        color={{ base: 'red', lg: 'green' }}
        x={{ base: 0, md: 100, lg: 200 }}
        hoverStyle={{
          backgroundColor: {
            base: '$backgroundHover',
            md: '$backgroundHoverStronger',
            lg: '$backgroundHoverStrongest',
          },
          x: { base: 0, md: 100, lg: 200 },
        }}
      />
    </Nested>
  )
}
```

Notice it's very hard to figure out what's happening here at any given combination.

Meanwhile, separating them out you very clearly see what's happening at each size. It also really isn't much more in terms of vertical space.

```tsx

export default () => {
  return (
    <Nested>
      <Component
        // here's everything you need to know for base size...
        width="50%"
        flexDirection="row"
        backgroundColor="$background"
        color="red"
        x={0}
        hoverStyle={{ backgroundColor: '$backgroundHover', x: 0 }}
        // ... now you can see what gets overridden at medium
        $md={{
          width: '43%',
          flexDirection: 'column',
          backgroundColor: '$backgroundStronger',
          hoverStyle: { backgroundColor: '$backgroundHoverStronger' }
        }}
        // ... now you can see what gets overridden at large
        $lg={{
          width: '35%',
          flexDirection: 'column-reverse',
          backgroundColor: '$backgroundStrongest',
          color: 'green',
          x: 200,
          hoverStyle: { backgroundColor: '$backgroundHoverStrongest' },
        }}
        // ... now you can see what gets overridden at xl
        $xl={{
          width: '28%',
          flexDirection: 'row-reverse',
        }}
      />
    </Nested>
  )
}
```

**Comment by seeden** [comment]
Hi @natew. Thank you for fast response.
I see your point. Here are my two cents:
1. You can skip items. It is a feature, not a bug. It is working like bootstrap (https://getbootstrap.com/docs/5.2/layout/grid/#stacked-to-horizontal) / material UI (https://mui.com/material-ui/react-grid/#basic-grid) "col / XS" attributes.

For example. Next object:
```js
{  sm: '43%', lg: '35%' }

```

Will be used as 
```js
{ base: 'inherti/default value', xs: 'inherit/default value', sm: '43%', md: '43%', lg: '35%', xl: '35%' }

```

2. In the production and real scenarios I am using "inline media query" most of the time just once. Therefore I do not have problem with combinations in one component. What I see more often is combination with other components which are using this inline media queries.
```js
<Box maxWidth={{ base: '600px', lg: 'auto'}} width="full">
  {children}
</Box>
<XStack width={{ base: '50%', md: '43%', lg: '35%', xl: '28%' }} space="layout">
```

3. I can live without it, but I will need to write more which can be confusing. As I know myself, I will try to use it anyway with "inline media query" in some hacky/dirty way

4. If somebody using combination in one component he can use your style.

**Comment by natew** [comment]
It’s maybe possible to actually hack this to work using variants and still get Tamagui to statically extract, but I havent tested that out at all and I’m pretty sure it would be 
 hard to get the types, compiler and runtime to all behave nicely. But could try it out.
